client/state/data-layer/wpcom/jetpack-blogs/product-install-status/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { JETPACK_PRODUCT_INSTALL_STATUS_REQUEST } from 'state/action-types';
+import { receiveJetpackProductInstallStatus } from 'state/jetpack-product-install/actions';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+
+/**
+ * Request the current Jetpack product install status.
+ *
+ * @param   {object} action Action to request the product install status.
+ * @returns {object}        The dispatched action.
+ */
+export const requestJetpackProductInstallStatus = action =>
+	http(
+		{
+			apiVersion: '1.1',
+			method: 'GET',
+			path: `/jetpack-blogs/${ action.siteId }/product-install-status`,
+		},
+		action
+	);
+
+/**
+ * Dispatches a product install status receive action when the request succeeded.
+ *
+ * @param   {Object} action Redux action
+ * @param   {Object} status Status as returned from the endpoint
+ * @returns {Object} Dispatched product install status receive action
+ */
+export const handleRequestSuccess = ( { siteId }, status ) =>
+	receiveJetpackProductInstallStatus( siteId, status );
+
+registerHandlers( 'state/data-layer/wpcom/jetpack-blogs/product-install-status', {
+	[ JETPACK_PRODUCT_INSTALL_STATUS_REQUEST ]: [
+		dispatchRequest( {
+			fetch: requestJetpackProductInstallStatus,
+			onSuccess: handleRequestSuccess,
+			onError: noop,
```

Comment:
We're relying on getting a response with like `{status: { progress: 42 } }`, where we use that number (0-100) with little or no protection elsewhere. What do you think about adding a `fromApi` to validate and sanitize some the incoming data?

Reply:
Good call - we don't really transform the endpoint response at all, but I agree validation has some value here. So I've introduced schema validation in 1e2457144042c78d15f1edb89757b8cc635e50cf